Client/Transfers/factory.ts

```diff
@@ -0,0 +1,173 @@
+import assert from "assert"
+import * as dotenv from "dotenv"
+import * as model from "../../model"
+
+dotenv.config()
+
+const sourceAccount = process.env["accountModulrEur"]
+assert(sourceAccount)
+
+const basicRequest: model.TransferRequest = {
+	providerCode: "modulr" as model.ProviderCode,
+	providerSourceAccountId: sourceAccount,
+	amount: 1,
+	currency: "EUR",
+}
+
+const basicExpected = {
+	sourceAccount: {
+		id: expect.any(Number),
+		providerAccountId: expect.any(String),
+		currency: expect.any(String),
+		state: expect.stringMatching(/(ACTIVE)|(INACTIVE)|(CLOSED)|(DELETED)|(EXPIRED)|(PENDING)|(APPROVED)|(DECLINED)|(GENERATED)/),
+		friendlyName: expect.any(String),
+		balance: expect.any(Number),
+		accountType: expect.stringMatching(/(FUNDING)|(CARD)/),
+		updatedOn: expect.any(String),
+		createdOn: expect.any(String),
+		provider: {
+			id: expect.any(Number),
+			code: basicRequest.providerCode,
+			name: expect.any(String),
+			status: expect.stringMatching(/(ACTIVE)|(DELETED)|(INACTIVE)/)
+		},
+		organisation: {
+			code: expect.any(String),
+			name: expect.any(String),
+			status: expect.stringMatching(/(ACTIVE)|(DELETED)/)
+		}
+	},  
+	amount: expect.any(Number),
+	status: expect.stringMatching(/(PENDING)|(PENDING_FOR_DATE)|(PENDING_FOR_FUNDS)|(SETTLED)|(CANCELLED)|(ERROR_REJECTED)|(APPROVAL_PENDING)|(DECLINED)|(APPROVED)|(GENERATED)/),
+	createdDate: expect.any(String),
+	paymentDate: expect.any(String),
+	reference: expect.any(String),
+	providerCode: basicRequest.providerCode,
+	providerTransferId: expect.any(String),
+	scheduled: expect.any(Boolean),
+}
+
+export function forBeneficiary(): [model.TransferRequest, model.TransferResponse] {
+
+	const request: model.TransferRequest = {
+		...basicRequest,
+		beneficiaryId: process.env.beneficiaryModulr,
+	}
+
+	assert(request.currency)
+
+	return [
+		request,
+		{
+			...basicExpected,
+			amount: request.amount,
+			beneficiary: {
+				transferDestination: {
+					currency: request.currency,
+					fullName: expect.any(String), 
+				},
+				defaultReference: expect.any(String),
+				status: expect.stringMatching(/(ACTIVE)|(DELETED)|(OUTDATED)/),
+				fullName: expect.any(String),
+				beneficiaryId: process.env.beneficiaryModulr,
+				createdOn: expect.any(String),
+			},
+
+			
+		}
+	]
+}
+
+export function forDestination(): [model.TransferRequest, model.TransferResponse, model.TransferRequest, model.TransferResponse] {
```

Comment:
as we discussed it might be nicer to have these as four separate methods, it'd be a lot clearer when using it what's meant to be what